vulcanai/models/metrics.py

```diff
@@ -787,6 +769,165 @@ def cross_validate(network, data_loader, k, epochs,
             kwargs: dict of keyworded parameters
                 Values passed to transform callable (function parameters)
 
+        Returns:
+            p_value : float
+        """
+        ls_imprv_scores = []
+
+        rand_sample = data.RandomSampler(data_loader.dataset, replacement=True)
+        data_loader_args = inspect.signature(data.DataLoader.__init__)
+        new_params = {}
+
+        for param in data_loader_args.parameters:
+            if param == "self":
+                continue
+            elif param == "shuffle":
+                continue
+            elif param == "sampler":
+                new_params["sampler"] = rand_sample
+            elif param == "batch_sampler":
+                new_params["batch_sampler"] = None
+            else:
+                new_params[param] = getattr(data_loader, param)
+
+        dl_sample = data.DataLoader(**new_params)
+
+        for samp in range(n_samples):
+            imprv_score = Metrics.boot_cv(network, dl_sample, k, epochs, retain_graph, 
+                valid_interv, plot, save_path, index_to_iter, ls_feat_vals)
+            ls_imprv_scores.append(imprv_score)
+
+        tot_num_imprv = float(len(ls_imprv_scores))
+        score_abv_zero = sum(val > 1.0 for val in ls_imprv_scores)
+        p_val = float(score_abv_zero)/float(tot_num_imprv)
```

Comment:
Wait I just noticed something. Shouldn't p_val = float(score_blw_zero)/float(tot_num_imprv)? Because score <= 0 means our model had no improvment.

Reply:
Yeah I just checked, it should be. My mistake I'll fix now.

Reply:
I fixed it and pushed. It would've been either way, just had to be interpreted differently than what we did for v1. Figured I would just change to how we did it before so it 'standardized' across both. So now, we have it calculating p value based on number of times we didn't see improvement.

Reply:
awesome thanks for keeping it consistent